flink-scala/src/main/scala/org/apache/flink/api/scala/unfinishedKeyPairOperation.scala

```diff
@@ -169,13 +165,10 @@ private[flink] class HalfUnfinishedKeyPairOperation[L, R, O](
    * the finished operation.
    */
   def equalTo[K: TypeInformation](fun: KeySelector[R, K]): O = {
+
     val keyType = implicitly[TypeInformation[K]]
-    val keyExtractor = new KeySelector[R, K] {
-      val cleanFun = unfinished.leftInput.clean(fun)
-      def getKey(in: R) = cleanFun.getKey(in)
-    }
     val rightKey = new Keys.SelectorFunctionKeys[R, K](
-      keyExtractor,
+      unfinished.leftInput.clean(fun),
```

Comment:
Should this be `unfinished.rightInput.clean(fun)`?

Reply:
It doesn't matter, I just need to somehow clean the function.